tests/sender/send_data.py

```diff
@@ -315,3 +322,228 @@ def test_sender_as_handler_static(self):
+                                     verify_mode=CERT_NONE,
+                                     verify_config=True)
+
+        wrong_cert_message = "Error in the configuration, " + \
+                             wrong_cert.cert + \
+                             " path does not exist"
+
+        wrong_chain = SenderConfigSSL(address=(self.server, self.port),
+                                      key=self.chain,
+                                      cert=self.cert,
+                                      chain="Incorrect chain",
+                                      check_hostname=False,
+                                      verify_mode=CERT_NONE,
+                                      verify_config=True)
+        wrong_chain_message = "Error in the configuration, " + \
+                              wrong_chain.chain + \
+                              " path does not exist"
+        test_params = [
+            (wrong_key, wrong_key_message),
+            (wrong_cert, wrong_cert_message),
+            (wrong_chain, wrong_chain_message),
+        ]
+
+        for config, expected_result in test_params:
+
+            with self.assertRaises(DevoSenderException) as result:
+                Sender.check_config_files_path(config)
+
+            self.assertEqual(expected_result, str(result.exception))
+
+    def test_config_cert_key_standard_case(self):
+        """
+        Test that verifies that a compatible certificate
+        and key are detected.
+        """
+
+        engine_config = SenderConfigSSL(address=(self.server, self.port),
+                                        key=self.key,
+                                        cert=self.cert,
+                                        chain=self.chain,
+                                        check_hostname=False,
+                                        verify_mode=CERT_NONE,
+                                        verify_config=True)
+        result = Sender.check_config_certificate_key(engine_config)
+        self.assertEqual(result, True)
+
+    def test_config_cert_key_incompatible_case(self):
+        """
+        Test that verifies that an incompatible
+        certificate with a key raises an exception.
+        """
+
+        engine_config = SenderConfigSSL(address=(self.server, self.port),
+                                        key=self.key,
+                                        cert=self.cert,
+                                        chain=self.chain,
+                                        check_hostname=False,
+                                        verify_mode=CERT_NONE,
+                                        verify_config=False)
+        incompatible_cert = \
+            "{!s}/local_certs/keys/server/server_cert.pem" \
+            .format(os.path.dirname(os.path.abspath(__file__)))
+        engine_config.cert = incompatible_cert
+        with self.assertRaises(DevoSenderException) as result:
+            Sender.check_config_certificate_key(engine_config)
```

Comment:
**suggestion:** instead of creating the object and assigning a property afterwards, you could have passed it as a parameter directly.

> Note that in a previous comment I suggested you perform the validation inside the configuration class, so bear in mind that you should probably have a different approach to test these cases.